app/Http/Controllers/CompetitionsController.php

```diff
@@ -48,7 +61,8 @@ public function store(Request $request)
      */
     public function show($id)
     {
-        //
+        $comp = Competition::findOrFail($id);
```

Comment:
You can type-hint the `Competition` and the controller will [automatically resolve it](https://laravel.com/docs/5.2/routing#route-model-binding) based on the model's primary key.

```php
public function show(Competition $competition)
{
      return view('competitions.show')->with(compact('competition'));
}
```